mev-build-rs/src/reth_builder/reth_compat.rs

```diff
@@ -22,7 +30,10 @@ pub(crate) fn to_u256(value: &U256) -> ssz_rs::U256 {
     ssz_rs::U256::try_from_bytes_le(&value.to_le_bytes::<32>()).unwrap()
 }
 
-pub(crate) fn to_execution_payload(value: &SealedBlock) -> ExecutionPayload {
+pub(crate) fn to_execution_payload<Pool>(value: &SealedBlock, pool: &Pool) -> ExecutionPayload
```

Comment:
it seems better to just pass in the fork schedule (perhaps just mapping from timestamp to fork) and then dispatch to the relevant type?

Reply:
I fixed this by fetching the currnet_epoch and comparing with the differents fork's epochs